### PR TITLE
[feat] create big data dropdown

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Added `BigDataDropdown` component

Modified `InputDropdown` styles (linked to `BigDataDropdown`)

Introduced new dependencies:
- `react-select-async-paginate`
- `iso-639-3`
- `country-state-city`

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

bigdatadropdown

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

#### Code Review

Review `components/BigDataDropdown/index.tsx`, then go in any order

#### Testing

Standard procedure:
```
git fetch origin jinkang/ijp-80-create-big-data-dropdowns
git checkout jinkang/ijp-80-create-big-data-dropdowns
```

Copy paste the example usage at the bottom of `components/BigDataDropdown/index.tsx` into a random test page, then start the app with `npm run dev`. Navigate to the test page to test it out.

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar, @davidqing6432

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,13 +17,16 @@
     "@types/node": "20.6.3",
     "@types/react": "18.2.22",
     "@types/react-dom": "18.2.7",
+    "country-state-city": "^3.2.1",
     "dotenv": "^16.3.1",
     "eslint-config-next": "13.5.2",
     "eslint-import-resolver-typescript": "^3.6.1",
+    "iso-639-3": "^3.0.1",
```
Comment (pragyakallanagoudar) [inline package.json:24]: i wonder if this package is TOO comprehensive...🤠 there are just so many variants that typing in something seemingly simple like "English" or "French" gives you _so many_ options

File: package.json [hunk 2]
```diff
@@ -17,13 +17,16 @@
     "@types/node": "20.6.3",
     "@types/react": "18.2.22",
     "@types/react-dom": "18.2.7",
+    "country-state-city": "^3.2.1",
```
Comment (pragyakallanagoudar) [inline package.json:20]: will ask ambreen if we can limit to U.S. cities for now

File: src/components/BigDataDropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+      options instanceof Set
+        ? Array.from(options).map(v => ({ label: v, value: v }))
+        : Array.from(options.entries()).map(([k, v]) => ({
+            value: k,
+            label: v,
+          })),
+    [options],
+  );
+
+  const handleChange = useCallback(
```
Comment (pragyakallanagoudar) [inline src/components/BigDataDropdown/index.tsx:71]: can we make it so that when something is typed into the search bar, we auto-scroll to the top? (not sure if this is the correct functionality? cc @davidqing6432)

File: package.json (hunk 1 above)
Comment (jinkang-0) [reply package.json:24]: lowk maybe [639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) is better

File: package.json (hunk 2 above)
Comment (jinkang-0) [reply package.json:20]: if not, we should consider fields for `country` and `state`/`province`

File: src/components/BigDataDropdown/index.tsx (hunk 3 above)
Comment (jinkang-0) [reply src/components/BigDataDropdown/index.tsx:71]: i think it makes sense, typically the most accurate results float to the top

File: package.json (hunk 1 above)
Comment (jinkang-0) [reply package.json:24]: it doesn't look like iso 639-1 support sign languages though, i'll look for an alternative

Comment (jinkang-0) [reply package.json:24]: ISO 639-2 provides a good comprehensive list that's not too overwhelming, but unfortunately also doesn't have sign languages. Maybe we can keep a table in Supabase to keep a list of languages that we want? Likely stretch

Comment (jinkang-0) [reply package.json:24]: For now, I have it as ISO 639-2 plus languages that end with "Sign Language"

File: src/components/BigDataDropdown/index.tsx (hunk 3 above)
Comment (jinkang-0) [reply src/components/BigDataDropdown/index.tsx:71]: i implemented this feature for both the `BigDataDropdown` and the `InputDropdown`, but will leave this open for david's comments

File: package.json (hunk 1 above)
Comment (pragyakallanagoudar) [reply package.json:24]: ok wonderful! tbh,,, this also seems like a LOT of languages lol

question: thoughts on removing `package.json` changes from this PR and just merging in the big data dropdowns? we could handle the actual package addition in the basic info card sprint since we're not 100% sure which ones we'll be using yet...but also we can also go back and change so no biggie to keep these

File: src/components/BigDataDropdown/index.tsx (hunk 3 above)
Comment (pragyakallanagoudar) [reply src/components/BigDataDropdown/index.tsx:71]: sounds good!!

**Review by pragyakallanagoudar — APPROVED** [review]
AMAZING WORK JINKANG!! mad props to you for figuring out the pagination stuff, this is so cool :))) 

imo this is good to go! there's still some questions about data sources for languages & cities, which will hopefully be cleared up after poc meeting tmr :))

File: package.json (hunk 1 above)
Comment (jinkang-0) [reply package.json:24]: maybe ISO 639-1 plus sign languages? But I think rn it's decently comprehensive without being TOO overwhelming. If people want a specific language, they can search it, and they won't have to drive down to a specific kind of English.

Comment (jinkang-0) [reply package.json:24]: for `package.json` - could drop after testing, but also not sure if there are better libraries that provide a good list of languages out there. I think we should keep it if we can't find anything else for now so *ideally* we don't have to edit package.json later.
